### PR TITLE
Heedls 541c edit filters again

### PR DESCRIPTION
**Pull request by JBsoftwire**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-541

### Description
This is the frontend component for the edit filters page for HEEDLS-541. It has some backend changes to populate the select lists, and to enable the uncapped end date filter, but no logic to handle actually sending a filter edit request, which will be dealt with in a future review. It also makes some changes to the Course Delegates controller and services to remove duplication and improve an SQL query.

### Screenshots
image
image

-----
### Developer checks
I have:
- [X] Run the formatter and made sure there are no IDE errors.
- [X] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [X] Updated/added documentation in Swiki and/or Readme.
- [X] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [X] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/ViewComponents/SelectListViewComponent.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class SelectListViewComponent : ViewComponent
             var hasError = ViewData.ModelState[property?.Name]?.Errors?.Count > 0;
             var errorMessage = hasError ? ViewData.ModelState[property?.Name]?.Errors[0].ErrorMessage : null;
 
-            var textBoxViewModel = new SelectListViewModel(
+            var selectListViewModel = new SelectListViewModel(
```
Comment (JBsoftwire) [inline DigitalLearningSolutions.Web/ViewComponents/SelectListViewComponent.cs:28]: Not related to my ticket, but I thought I should fix this in passing.

**Review by ibrahimmunir14 — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Data/Models/Courses/CourseNameInfo.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 {
     public class CourseNameInfo : BaseSearchableItem
     {
+        public int CustomisationId { get; set; }
```
[the remark this answers is not in the document]
Comment (JBsoftwire) [reply DigitalLearningSolutions.Data/Models/Courses/CourseNameInfo.cs:5]: Holdover from before the 'Course' object was introduced, whoops.

File: DigitalLearningSolutions.Web/Views/Shared/Components/DateRangeInput/Default.cshtml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    @if (Model.HasError)
+    {
+      <span class="error-message--margin-bottom-1 nhsuk-error-message">
+        <span class="nhsuk-u-visually-hidden">Error:</span> @Model.ErrorMessage
+      </span>
+    }
+
+    @if (Model.EndDateCheckbox != null) {
```
[the remark this answers is not in the document]
Comment (JBsoftwire) [reply DigitalLearningSolutions.Web/Views/Shared/Components/DateRangeInput/Default.cshtml:35]: It is for this, but I wanted to leave in the option of not including it for the generic component. However this does require updating the view model.

**Comment by JBsoftwire** [comment]
Added 'from' to start date after demo feedback.
image
image

**Pull request closed without merging**